spec/models/solr_document_spec.rb

```diff
@@ -124,6 +124,36 @@
     end
   end
 
+  describe '#library_display' do
+    solr_info = { id: 30, item_display_struct:
+                [{ barcode: '9102385731231', callnumber: nil, current_location: nil,
+                   effective_location_id: '43c011f3-adeb-4c49-b357-06772aff11de',
+                   full_shelfkey: nil, home_location: 'BUS-ELECTRONIC', library: 'BUSINESS',
+                   permanent_location_code: 'BUS-ELECTRONIC', scheme: 'LC', type: 'ONLINE' }] }
+    physical_location_hash = solr_info.deep_dup
+    physical_location_hash[:item_display_struct].push({ barcode: 123456, home_location: 'BUS', library: 'BUSINESS',
+                                                        permanent_location_code: 'BUS' })
+    let(:solr_doc) { SolrDocument.new(solr_info) }
+    let(:physical_location_doc) { SolrDocument.new(physical_location_hash) }
+
+    it 'is has a library_display' do
```

Comment:
can you make this one context block for `when it has holdings but no physical items`